packages/flutter_tools/lib/src/commands/format.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class FormatCommand extends FlutterCommand {
+  @override
+  final String name = 'format';
+
+  @override
+  final String description = 'Format one or more dart files.';
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  @override
+  String get invocation => "${runner.executableName} $name <one or more paths>";
+
+  @override
+  Future<int> runInProject() async {
+    if (argResults.rest.isEmpty) {
+      printStatus('No files specified to be formatted.');
+      printStatus(usage);
+      return 1;
+    }
+
+    String dartfmt = path.join(
+        Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartfmt');
+    List<String> cmd = <String>[dartfmt, '-w']..addAll(argResults.rest);
```

Comment:
One thing to consider with that is that we'll be referencing another package that depends on the analyzer package (https://github.com/dart-lang/dart_style/blob/master/pubspec.yaml#L10). The `analyzer` package is already at the nexus of some version lock issues, so this could be an opportunity to opt-out of more version conflicts.